ml/main.py

```diff
@@ -0,0 +1,152 @@
+
+  # combine food & nutrient data
+  combined = food.join(food_nutrient.set_index('fdc_id'))
+  combined = combined.join(nutrient.set_index('id'), on='nutrient_id')
+
+  # extract energy/kcal data
+  calorie_data = combined[(combined['name'] == 'Energy')
+                          & (combined['unit_name'] == 'KCAL')]
+  calorie_data = calorie_data[[
+      'description', 'data_type', 'name', 'amount', 'unit_name'
+  ]]  # keep relevant cols
+  calorie_data = calorie_data.dropna().drop_duplicates()  # clean data
+
+  return calorie_data
+
+
+def add_calorie_labels(calorie_data, low_calorie_threshold,
+                       high_calorie_threshold):
+  """Set 'calorie_label' for each row in the dataset to reflect which class of calorie that dish falls under"""
+
+  def label_row(calorie_data_row):
+    if calorie_data_row['amount'] < low_calorie_threshold:
+      return [1, 0, 0] # low calorie
+    elif calorie_data_row['amount'] > high_calorie_threshold:
+      return [0, 0, 1] # high calorie
+    else:
+      return [0, 1, 0] # avg calorie
+
+  calorie_data['calorie_label'] = calorie_data.apply(
+      lambda row: label_row(row), axis=1)
+
+  return calorie_data
+
+
+def num_unique_words(series):
+  """Counts the number of unique words (space-separated) in all strings of a pandas Series."""
+  words = set()
+  for entry in series:
+    words.update(entry.split(' +'))
+
+  return len(words)
+
+
+def max_sequence_length(series):
+  """Compute the max length of the values in a pandas Series."""
+  max_len = None
+  for entry in series:
+    if max_len is None or len(entry) > max_len:
+      max_len = len(entry)
+
+  return max_len
+
+def add_input_labels(calorie_data, vocab_size, max_corpus_length):
+  """Set 'input' for each row in the dataset to reflect the encoded input of the text."""
+
+  def input_for_row(row):
+    encoded = keras.preprocessing.text.one_hot(row['description'], vocab_size)
+    
+    while len(encoded) < max_corpus_length:
+      encoded.append(0)
+
+    return encoded
+
+  calorie_data['input'] = calorie_data.apply(lambda row: input_for_row(row), axis=1)
+  return calorie_data
+
+
+# %% read data from disk
+print(f'Tensorflow version {tf.__version__}')
+calorie_data = load_calorie_data()
+add_calorie_labels(calorie_data, low_calorie_threshold=100, high_calorie_threshold=500)
+
+# %% add encode 'description' into a new 'input' column in calorie_data
+calorie_data['description'] = calorie_data['description'].str.replace(',', '').str.lower()
+corpus = calorie_data['description']
+vocab_size = num_unique_words(corpus)
+tokenized_corpus = corpus.map(lambda desc: desc.split(' '))
+max_corpus_length = max_sequence_length(tokenized_corpus)
+calorie_data = add_input_labels(calorie_data, vocab_size, max_corpus_length)
+
+# %% create model
+model = keras.Sequential([
+  keras.layers.Embedding(vocab_size, int(vocab_size**(1/4)), input_length=max_corpus_length),
+  keras.layers.Flatten(),
```

Comment:
Probably want more intermediate layers here, yeah?

Reply:
Added a couple Dense layers with sigmoid activation in https://github.com/googleinterns/amaranth/pull/2/commits/db41980dda44b6d0bc3acd54f3ce27d7935a9e54:

https://github.com/googleinterns/amaranth/blob/db41980dda44b6d0bc3acd54f3ce27d7935a9e54/ml/main.py#L72-L79

However, the models performance didn't change by more than a percentage point in precision and recall. Is there some other layer type/activation function/layer config that I should try? I was just playing around with different values and things, not sure if there's any kind of rule of thumb